examples/tv_minimization_reconstruction_pdhg.py

```diff
@@ -0,0 +1,184 @@
+#
+# PDHG essentially consists of three steps, which read as
+#
+# $z_{k+1} = \mathrm{prox}_{\sigma f^{\ast}}(z_k + \sigma K \bar{x}_k)$
+#
+# $x_{k+1} = \mathrm{prox}_{\tau g}(x_k - \tau K^H z_{k+1})$
+#
+# $\bar{x}_{k+1} = x_{k+1} + \theta(x_{k+1} - x_k)$,
+#
+# where $\mathrm{prox}$ denotes the proximal operator and $f^{\ast}$ denotes the convex conjugate of the
+# functional $f$, $\theta\in [0,1]$ and step sizes $\sigma, \tau$ such that $\sigma \tau < 1/L^2$, where
+# $L=\|K\|_2$ is the operator norm of the operator $K$.
+#
+# The first step is to recast problem (1) into the general form of (2) and then to apply the steps above
+# in an iterative fashion.
+#
+# A possible and intuitive (but unfortunately not efficient) identification is the following
+#
+# $f(x) = \lambda \| x\|_1,$
+#
+# $g(x) = \frac{1}{2}\|Ax  - y\|_2^2,$
+#
+# $K(x) = \nabla x.$
+#
+# However, although $\mathrm{prox}_{\sigma f^\ast}$ has a simple form, calculations show that
+# to be able to compute $\mathrm{prox}_{\tau g}$, one would need to solve a linear system at each
+# iteration. We will therefore use a way more efficient identification.
+#
+# In the following, we load some 2D radial MR data and PDHG to reconstruct it.
+
+# %%
+import mrpro
+
+# %% [markdown]
+# ##### Read-in the raw data
+
+# %%
+# Load in the Data from the ISMRMRD file
+kdata = mrpro.data.KData.from_file(data_file.name, mrpro.data.traj_calculators.KTrajectoryIsmrmrd())
+kdata.header.recon_matrix.x = 256
+kdata.header.recon_matrix.y = 256
+
+# %% [markdown]
+# ### Set up the operator $A$
+# Estimate coil sensitivity maps and density compensation function. Also run a direct (adjoint)
+# reconstruction and iterative SENSE as methods of comparison.
+
+# %%
+# Set up direct reconstruction class to estimate coil sensitivity maps, density compensation etc
+direct_reconstruction = mrpro.algorithms.reconstruction.DirectReconstruction(kdata)
+img_direct = direct_reconstruction(kdata)
+
+# run tierative SENSE to get an initiual guess of the solution
+iterative_sense_reconstruction = mrpro.algorithms.reconstruction.IterativeSENSEReconstruction(
+    kdata,
+    n_iterations=8,
+    csm=direct_reconstruction.csm,
+    dcf=direct_reconstruction.dcf,
+)
+img_iterative_sense = iterative_sense_reconstruction(kdata)
+
+# Define Fourier operator and csm-operator re-using the ones already constructed in the
+# direct reconstruction class
+fourier_operator = direct_reconstruction.fourier_op
+assert direct_reconstruction.csm is not None
+csm_operator = direct_reconstruction.csm.as_operator()
+
+# Create the entire acquisition operator A to be used in the operator K=[A, \nabla]^T for PDHG
+acquisition_operator = fourier_operator @ csm_operator
+
+# adjoint reconstruction as additional method of comparison
+(img_adjoint,) = acquisition_operator.H(kdata.data)
+
+
+# %% [markdown]
+# ### Recast the problem to be able to apply PDHG
```

Comment:
See comment above - I would move this to the intro

Reply:
same as above... I am fine with either with maybe a slight preference of having first the text and then all the code